src/app/api/services/paginator/paginator.service.ts

```diff
@@ -0,0 +1,203 @@
+            this.setLinks(response.header(PaginatorService.LINK_HEADER_NAME));
+            this.paginationPerPage = parseInt(response.header(PaginatorService.PAGINATION_PER_PAGE_HEADER_NAME), 10);
+            this.currentPage = parseInt(response.header(PaginatorService.PAGINATION_CURRENT_PAGE_HEADER_NAME), 10);
+            this.maxLinksPage = parseInt(response.header(PaginatorService.PAGINATION_MAX_LINK_PAGES_HEADER_NAME), 10);
+            this.totalRecords = parseInt(response.header(PaginatorService.PAGINATION_TOTAL_ENTRIES_HEADER_NAME), 10);
+            return response.entity;
+        });
+    }
+
+    /**
+     * Request the last page
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getLastPage(): Observable<any[]> {
+        return this.get(this.links.last);
+    }
+
+    /**
+     * Request the first page
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getFirstPage(): Observable<any[]> {
+        return this.get(this.links.first);
+    }
+
+    /**
+     * request the  pageParam page
+     * @param {number} [pageParam=1] Page to request
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorServic
+     */
+    public getPage(pageParam = 1): Observable<any[]> {
+        let page = this.links['x-page'] ? this.links['x-page'].replace('{pageValue}', String(pageParam))
+                        : undefined;
+        return this.get(page);
+    }
+
+    /**
+     * Request the next page
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getNextPage(): Observable<any[]> {
+        return this.get(this.links.next);
+    }
+
+    /**
+     * Request the previous page
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getPrevPage(): Observable<any[]> {
+        return this.get(this.links.prev);
+    }
+
+    /**
+     * Use the offset to request a page.
+     * @param {number} offset Offset to be request
+     * @returns {Observable<any[]>}
+     * @memberof PaginatorService
+     */
+    public getWithOffset(offset: number): Observable<any[]> {
+        let page = this.getPageFromOffset(offset);
+        return this.getPage(page);
+    }
+
+    private getPageFromOffset(offset: number): number {
+        return parseInt(String(offset / this.paginationPerPage), 10) + 1;
+    }
+
+    private setLinks(linksString: string): void {
+        let linkSplit = linksString.split(',');
+
+        linkSplit.forEach(linkRel => {
+            let linkrealSplit = linkRel.split(';');
+            let url = linkrealSplit[0].substring(1, linkrealSplit[0].length - 1);
+            let relSplit = linkrealSplit[1].split('=');
+            let rel = relSplit[1].substring(1, relSplit[1].length - 1);
+            this.links[rel] = url;
+        });
+    }
+}
+
+interface Links {
+    first?: string;
+    last?: string;
+    next?: string;
+    'x-page'?: string;
```

Comment:
should be use this like x-page, probably just xPage would be more standard and easier, if there is not any special case to use it as a 'x-page'